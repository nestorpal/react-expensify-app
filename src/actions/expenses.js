import uuid from 'uuid';

// ADD_EXPENSE
const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {}) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE EXPENSE
const removeExpense = ({ id }) => ({
    type: 'REMOVE_EXPENSE',
    id
});

// EDIT EXPENSE
const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

export { addExpense, removeExpense, editExpense }