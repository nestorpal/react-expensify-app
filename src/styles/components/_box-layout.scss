.box-layout {
    align-items: center;
    background-size: cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white, .15);
    border-radius: 3px;
    padding: $large-size $medium-size;
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $medium-size 0;
    line-height: 1;
}