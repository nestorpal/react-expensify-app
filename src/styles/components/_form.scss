.form {
    display: flex;
    flex-direction: column;

    > * { // affects direct children. Doesn't consider elements inside child div
        margin-bottom: $medium-size;
    }
}

.form__error {
    font-style: italic;
    margin: 0 0 $medium-size 0;
}