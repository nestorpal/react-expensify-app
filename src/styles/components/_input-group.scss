.input-group {
    display: flex; // puts things left to right
    flex-direction: column;
    margin-bottom: $medium-size;

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $large-size;
    }
}

.input-group__item {
    margin-bottom: $small-size;

    @media (min-width: $desktop-breakpoint) {
        margin: 0 $small-size 0 0;
    }
}