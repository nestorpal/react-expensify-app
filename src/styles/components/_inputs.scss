.text-input {
    border: 1px solid #cacccd;
    font-size: $font-size-large;
    font-weight: 300;
    height: 50px;
    padding: $small-size;
}

.select {
    @extend .text-input;
    
}

.text-area {
    @extend .text-input;
    height: 10rem;
}