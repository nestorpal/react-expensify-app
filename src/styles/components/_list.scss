.list-header {
    background: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $gray;
    display: flex;
    justify-content: space-between;
    padding: $small-size $medium-size;
}

.list-body {
    margin-bottom: $medium-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $large-size;
    }
}

.list-item {
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-gray;
    display: flex;
    flex-direction: column;
    padding: $small-size;
    text-decoration: none;
    transition: background .2s ease;
    &:hover {
        background: $off-white;
    }

    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $medium-size;
    }
}

.list-item--message {
    align-items: center;
    color: $gray;
    justify-content: center;
    padding: $medium-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__sub-title {
    color: $gray;
    font-size: $font-size-small;
}

.list-item__data {
    margin: $small-size 0 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $small-size;
    }
}