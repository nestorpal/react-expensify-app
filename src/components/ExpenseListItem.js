import React from 'react';
import { NavLink } from 'react-router-dom';

import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <NavLink to={`/edit/${id}`}>
            <h3>{description}</h3>
        </NavLink>
        <p>{amount} - {createdAt}</p>
    </div>
);

export default ExpenseListItem;