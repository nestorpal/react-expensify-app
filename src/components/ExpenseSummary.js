import React from 'react';
import { connect } from 'react-redux';
import expenses from '../selectors/expenses';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    return (
        <div>
            <p>
                {`Viewing ${expenseCount} ${(expenseCount != 1 ? ' expenses' : ' expense' )} totalling ${numeral(expensesTotal / 100).format('$0,0.00')}`}
            </p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);