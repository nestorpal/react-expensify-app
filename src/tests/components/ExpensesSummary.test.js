import React from 'react';
import { ExpensesSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';

test('Should render Expenses Summary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
});

test('Should render Expenses Summary correctly 2', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
});