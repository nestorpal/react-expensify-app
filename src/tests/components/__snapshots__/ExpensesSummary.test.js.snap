// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expenses Summary correctly 1`] = `
<div>
  <p>
    Viewing undefined  expenses totalling $0.00
  </p>
</div>
`;

exports[`Should render Expenses Summary correctly 2 1`] = `
<div>
  <p>
    Viewing undefined  expenses totalling $0.00
  </p>
</div>
`;
